Core/Src/mcp3008.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * MCP3008.c
+ *
+ */
+
+#include "../Inc/mcp3008.h"
+
+/*
+ * Set the MISO, MOSI, SCK and CS
+ * SPI settings:
+ * CPHA = 1 Edge
+ * Prescaler = 8
+ * First bit = MBS first
+ * CPOL = Low
+ */
+void MCP3008_Init(MCP3008_SPI* spi, SPI_HandleTypeDef* hspi, GPIO_TypeDef* CS_PORT, uint16_t CS_PIN){
+  spi->hspi = hspi;
+  spi->CS_PORT = CS_PORT;
+  spi->CS_PIN = CS_PIN;
+}
+
+// Read the channels from 0 to 7
+uint16_t MCP3008_Read_Channel(MCP3008_SPI* spi, uint8_t channel){
+
+	// Declare data that we will send
+	uint8_t pTxData[3] = {0};
+	pTxData[0] = (0x01 << 7);		  // start bit
+
+	pTxData[1] = (1 << 7)|			// SGL
+				 ((channel & 0x07) << 4); 	// channel number, channel is just 0 so this line will evaluate to 0 either way

```

Comment:
you are almost there
![image](https://github.com/UWARG/embedded-bootcamp/assets/106219057/64f0ed0f-3fd7-407b-a1a5-8d9b5092c510)
the SGL is the least significant bit, channel number is the 3 bits after SGL

Reply:
that seems so backwards from the diagram, does that mean my pRxData will be wrong as well?

Reply:
least significant bit is usually on the right, right?

Reply:
your rx data is actually right.
The position of LSB is very context dependent. In this datasheet, it starts the data transmission from left to right. 
![image](https://github.com/UWARG/embedded-bootcamp/assets/106219057/7eb28eda-a915-4c13-9f71-54bfc44311f1)
This is also proving the start bit is the MSB of the first tx byte

Reply:
in that case, I think it should be good now